### PR TITLE
Added GitHub release workflow and docs builds

### PR DESCRIPTION
**Pull request by davidwaroquiers**
Copied and modified the release workflow from qtoolkit.
Need @ml-evs to check it is fine.
Maybe it would be good to add the release procedure (when, how, who, ...) in the dev documentation some place.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        python-version: '3.11'
+
+    - name: Install Python dependencies
+      run: |
+        python -m pip install -U pip
+        pip install -U setuptools wheel
+        pip install -e .[all]
+
+    - name: Update changelog
+      uses: CharMixer/auto-changelog-action@v1
```
Comment (ml-evs) [inline .github/workflows/release.yml:45]: Might have to delete the contents of the current changelog before this works usefully -- can also have a workflow in the tests that adds to the "Unreleased" section of the changelog every time you merge into `main`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      with:
+        user: __token__
+        password: ${{ secrets.TEST_PYPI_PASSWORD }}
+        repository-url: https://test.pypi.org/legacy/
+
+    - name: Install docs dependencies
+      run: |
+        # Required to generate rst files from markdown
+        sudo apt install pandoc
+        pip install .[docs]
```
Comment (ml-evs) [inline .github/workflows/release.yml]: This is fine, but I would suggest building the docs before releasing and then only uploading after the release; this should prevent needing to re-release if the docs break in some unpredictable way

**Review by ml-evs — COMMENTED** [review]
LGTM @davidwaroquiers, just a couple of comments below.

As is always the case with these things, the only problems will come from having the right permissions rather than anything else...

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      uses: CharMixer/auto-changelog-action@v1
+      with:
+        token: ${{ secrets.RELEASE_PAT_BOT }}
+        release_branch: ${{ env.PUBLISH_UPDATE_BRANCH }}
+        exclude_labels: "duplicate,question,invalid,wontfix,dependency_updates,skip_changelog"
+
+    - name: Update '${{ env.PUBLISH_UPDATE_BRANCH }}'
+      uses: CasperWA/push-protected@v2
+      with:
+        token: ${{ secrets.RELEASE_PAT_BOT }}
```
Comment (ml-evs) [inline .github/workflows/release.yml:54]: For example, I don't see this secret in the repo atm

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    - name: Build source distribution
+      run: |
+        pip install -U build
+        python -m build
+
+    - name: Publish package to Test PyPI first
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        user: __token__
+        password: ${{ secrets.TEST_PYPI_PASSWORD }}
```
Comment (ml-evs) [inline .github/workflows/release.yml:114]: nor this one (sorry for breaking up the review)

File: .github/workflows/release.yml (hunk 2 above)
Comment (davidwaroquiers) [reply .github/workflows/release.yml]: Ok I'll change that. I took it as such from qtoolkit, I will also modify it there then.

File: .github/workflows/release.yml (hunk 3 above)
Comment (davidwaroquiers) [reply .github/workflows/release.yml:54]: Indeed, for qtoolkit, I think you added your own secrets there if I remember correctly ? Rationale was that e.g. pypi does not (yet, it is under "closed beta" as they say) organization accounts. Was it indeed what we did ? Are you ok to add your secret there ? (what does PAT mean btw ?)

File: .github/workflows/release.yml (hunk 4 above)
Comment (davidwaroquiers) [reply .github/workflows/release.yml:114]: Same here, are you ok to add your secret ?

File: .github/workflows/release.yml (hunk 3 above)
Comment (ml-evs) [reply .github/workflows/release.yml:54]: I can generate one and add it, sure!

PAT = Personal Access Token

Comment (ml-evs) [reply .github/workflows/release.yml:54]: Now is probably the time to also make a bot account for Matgenix, I will do that and use the token for it here. The email will be dev@matgenix.com, which presumably someone is watching?

File: .github/workflows/release.yml (hunk 4 above)
Comment (ml-evs) [reply .github/workflows/release.yml:114]: Yep, just bear in mind that we don't have the "real" release workflow in this one yet

File: doc/source/conf.py [hunk 5]
```diff
@@ -21,7 +21,7 @@
 
 # -- Project information -----------------------------------------------------
 
-project = "Jobflow-Remote"
+project = "Jobflow Remote"
 copyright = "2023, Matgenix SRL"
 author = "Guido Petretto, David Waroquiers"
```
Comment (davidwaroquiers) [inline doc/source/conf.py]: Matthew Evans

**Pull request merged**